src/MerkleDistributor.sol

```diff
@@ -60,6 +61,20 @@ contract MerkleDistributor is Ownable2Step, EIP712, IMerkleDistributor {
         _transferOwnership(_initialOwner);
     }
 
+    /**
+     * @notice Reverts if called by any account other than an enabled distributor.
+     */
+    modifier onlyDistributor() {
+        _checkDistributor();
+        _;
+    }
+
+    function _checkDistributor() internal view {
```

Comment:
Don't think we should extract the check to the separate method, lets keep the logic within modifier